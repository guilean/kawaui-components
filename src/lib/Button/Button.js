import './Button.scss';

import PropTypes from 'prop-types';
import React from 'react';
import clsx from 'clsx';

const Button = ({
	children = null,
	disabled = false,
	startIcon = undefined,
	endIcon = undefined,
	fullWidth = false,
	size = 'medium',
	onClick = undefined,
}) => {
	return (
		<button
			className={clsx('btn', `btn--${size}`, {
				['btn--fullWidth']: fullWidth,
			})}
			onClick={onClick}
			disabled={disabled}
		>
			{startIcon && startIcon}
			{children}
			{endIcon && endIcon}
		</button>
	);
};

Button.propTypes = {
	children: PropTypes.node.isRequired,
	disabled: PropTypes.bool,
	startIcon: PropTypes.node,
	endIcon: PropTypes.node,
	fullWidth: PropTypes.bool,
	size: PropTypes.string,
	onClick: PropTypes.func,
};

export default Button;
