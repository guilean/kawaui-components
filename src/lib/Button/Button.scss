.btn {
	color: red;
	&--small {
		width: 60px;
	}
	&--medium {
		width: 80px;
	}
	&--large {
		width: 120px;
	}
	&--fullWidth {
		width: 100%;
	}
}
