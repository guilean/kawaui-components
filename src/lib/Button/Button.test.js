import { cleanup, fireEvent, render } from '@testing-library/react';

import Button from './Button';
import React from 'react';

describe('<Button/>', () => {
	afterEach(() => {
		cleanup();
	});

	it('snnapshot', () => {
		const { getByRole } = render(<Button children={'Dummy'} />);
		const component = getByRole('button');
		expect(component).toMatchSnapshot();
	});

	it('should render a button node element', () => {
		const { getByRole } = render(<Button children={'Dummy'} />);
		const component = getByRole('button');
		expect(component.nodeName).toBe('BUTTON');
	});

	it('should have default btn class when rendered', () => {
		const { getByRole } = render(<Button children={'Dummy'} />);
		const component = getByRole('button');
		expect(component).toHaveClass('btn');
	});

	it('should change class with size prop', () => {
		const { getByRole } = render(<Button children={'Dummy'} size="large" />);
		const component = getByRole('button');
		expect(component).toHaveClass('btn--large');
	});

	it('should have disabled attribute', () => {
		const { getByRole } = render(<Button children={'Dummy'} disabled />);
		const component = getByRole('button');
		expect(component.hasAttribute('disabled')).toBeTruthy();
	});

	it('should call onClick when button is clicked', () => {
		const mockFn = jest.fn();
		const { getByRole } = render(
			<Button onClick={mockFn} children={'Dummy'} />
		);
		const component = getByRole('button');
		fireEvent.click(component);
		expect(mockFn).toHaveBeenCalledTimes(1);
	});
});
