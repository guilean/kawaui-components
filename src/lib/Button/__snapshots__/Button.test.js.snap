// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> snnapshot 1`] = `
<button
  class="btn btn--medium"
>
  Dummy
</button>
`;
